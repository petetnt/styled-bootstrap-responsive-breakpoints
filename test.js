import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import "jest-styled-components";

import {
  breakpoints,

  mediaBreakpointUpSm,
  mediaBreakpointUpMd,
  mediaBreakpointUpLg,
  mediaBreakpointUpXl,

  mediaBreakpointDownXs,
  mediaBreakpointDownSm,
  mediaBreakpointDownMd,
  mediaBreakpointDownLg,

  mediaBreakpointOnlyXs,
  mediaBreakpointOnlySm,
  mediaBreakpointOnlyMd,
  mediaBreakpointOnlyLg,
  mediaBreakpointOnlyXl,

  mediaBreakpointBetween,
} from "./index";

const componentFactory = (mediaQuery) => {
  const colorBlue = "blue";

  const Wrapper = styled.section`
    padding: 4em;
    background: papayawhip;
  `;

  const Title = styled.h1`
    ${mediaQuery`
      color: ${colorBlue};
    `}
  `;

  const tree = renderer.create(
    <Wrapper>
      <Title>Hello World!</Title>
    </Wrapper>,
  ).toJSON();

  return tree;
};

test("It should export bootstrap breakpoints", () => {
  const {
    xs, sm, md, lg, xl,
  } = breakpoints;

  expect(xs).toEqual("0px");
  expect(sm).toEqual("576px");
  expect(md).toEqual("768px");
  expect(lg).toEqual("992px");
  expect(xl).toEqual("1200px");
});

test("It should match the mediaBreakpointUpSm snapshot", () => {
  const tree = componentFactory(mediaBreakpointUpSm);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointUpMd snapshot", () => {
  const tree = componentFactory(mediaBreakpointUpMd);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointUpLg snapshot", () => {
  const tree = componentFactory(mediaBreakpointUpLg);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointUpXl snapshot", () => {
  const tree = componentFactory(mediaBreakpointUpXl);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointDownXs snapshot", () => {
  const tree = componentFactory(mediaBreakpointDownXs);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointDowSm snapshot", () => {
  const tree = componentFactory(mediaBreakpointDownSm);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointDownMd snapshot", () => {
  const tree = componentFactory(mediaBreakpointDownMd);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointDownLg snapshot", () => {
  const tree = componentFactory(mediaBreakpointDownLg);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointOnlyXs snapshot", () => {
  const tree = componentFactory(mediaBreakpointOnlyXs);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointOnlySm snapshot", () => {
  const tree = componentFactory(mediaBreakpointOnlySm);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointOnlyMd snapshot", () => {
  const tree = componentFactory(mediaBreakpointOnlyMd);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointOnlyLg snapshot", () => {
  const tree = componentFactory(mediaBreakpointOnlyLg);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointOnlyXl snapshot", () => {
  const tree = componentFactory(mediaBreakpointOnlyXl);
  expect(tree).toMatchSnapshot();
});

test("It should match the mediaBreakpointBetween snapshot", () => {
  const tree = componentFactory(mediaBreakpointBetween(100, 200));
  expect(tree).toMatchSnapshot();
});
