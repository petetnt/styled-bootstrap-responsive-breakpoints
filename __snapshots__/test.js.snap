// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should match the mediaBreakpointBetween snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:100px) and (max-width:199px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointDowSm snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (max-width:767px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointDownLg snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (max-width:1199px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointDownMd snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (max-width:991px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointDownXs snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (max-width:575px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointOnlyLg snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:992px) and (max-width:1199px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointOnlyMd snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:768px) and (max-width:991px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointOnlySm snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:576px) and (max-width:767px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointOnlyXl snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:1200px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointOnlyXs snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (max-width:575px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointUpLg snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:992px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointUpMd snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:768px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointUpSm snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:576px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointUpXl snapshot 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

@media (min-width:1200px) {
  .c1 {
    color: blue;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Hello World!
  </h1>
</section>
`;
